ratelimiting/ratelimiting_user.c

```diff
@@ -309,8 +380,33 @@ int main(int argc, char **argv)
         return 1;
     }
 
+    unsigned int id = 0;
+    int prev_prog_map_fd, nb_fds = 0;
+    int err;
+    
+    while(id >= 0)
+    {
+        err = bpf_map_get_next_id(id, &id);
+        printf("Inside ebpf_api_get_pinned_map_info... id : %d\n",id);
+        prev_prog_map_fd = bpf_map_get_fd_by_id(id);
+        printf("Inside ebpf_api_get_pinned_map_info... fd : %d\n",prev_prog_map_fd);
+        struct bpf_prog_info info = {0};
+        uint32_t info_size = (uint32_t)sizeof(info);
+        if (bpf_obj_get_info_by_fd(prev_prog_map_fd, &info, &info_size) == 0) {
+            printf("%7u  bpf_obj_get_info_by_fd  %s\n", info.id, info.name);
+            char *mapstr;
+            mapstr = strstr(prev_prog_map, info.name);
+            if (mapstr) {
+                printf("Matched pin path breaking the loop : %s\n",prev_prog_map);
+                break;
+            }
+        }
+    }
+
     /* Get the previous program's map fd in the chain */
-    int prev_prog_map_fd = bpf_obj_get(prev_prog_map);
```

Comment:
bpf_obj_get(prev_prog_map) is failing so had to use the above while loop to fetch the prev_prog_map_fd